lib/frizz/remote.rb

```diff
@@ -39,7 +50,9 @@ def delete(remote_file)
     attr_reader :bucket_name, :ignorance, :options
 
     def ignore?(object)
-      ignorance.ignore?(object.key)
+      # for remote files, you don't need to ignore for gzip preference,
+      # since there won't be any gzip files on s3
+      ignorance.ignore_matched_pattern?(object.key)
```

Comment:
@mrkrlli thanks for all the detail. This is interesting and we might be introducing a new concept here: supporting mis-matched name mapping between local and remote. If we support it in this case for gz, we could come up with some design/logic to make that handling generic, allowing us to support it in other cases as well. In a system like that, an object in `Ignorance`'s role wouldn't need to even care about the gzip option, it would just need to know the name that it's checking for — and the calling context would determine what name to use from that local<->remote mapping. 

I'm saying all this before reviewing your latest changes just to get my ideas on paper. Let me step into the changes now.